Improve the Health API documentation page with recent additions (#87139)
elastic/elasticsearch: docs/reference/health/health.asciidoc

[[health-api-desc]]
==== {api-description-title}

The health API returns the health status of an Elasticsearch cluster. It
returns a list of components that compose Elasticsearch functionality. Each
component's health is determined by health indicators associated with the

> component.

Reviewer: Shall we be a bit more verbose here and mention impacts and user actions? (briefly describing all the basic concepts of the API)

Each indicator has a health status of: `green`, `unknown`, `yellow` or `red`. The indicator will
provide an explanation and metadata describing the reason for its current health status.

A component's status is controlled by the worst indicator status. The cluster's status
is controlled by the worst component status.

[[health-api-path-params]]
==== {api-path-parms-title}

`<component>`::
    (Optional, string) Limits the information returned to the specific component.
    A comma-separated list of the following options:
+
--
  `cluster_coordination`::

>       All health reporting that pertains to successful, stable, and continued membership
>       in a cluster.

Reviewer 2: ```suggestion
    All health reporting that pertains to a single, stable elected master which 
coordinates other nodes to ensure they all have a consistent view of the
<<cluster-state, cluster state>>.  If your cluster doesn’t have a stable
master, many of its features won’t work correctly. You must fix the master
node’s instability before addressing these other issues. It will not be
possible to solve any other issues while the master node is unstable.
```

Maybe someone from @elastic/es-distributed  team can help review the description on the cluster coordination? This text will also be used by UI via a ℹ️ tooltip to expand on the component. 

Text is taken from https://www.elastic.co/blog/a-new-era-for-cluster-coordination-in-elasticsearch and https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-discovery-hosts-providers.html

  `data`::
      All health reporting that pertains to data availability, lifecycle, and responsiveness.

  `snapshot`::
      All health reporting that pertains to data archival, backups, and searchable snapshots.
--

`<indicator>`::
    (Optional, string) Limit the information returned for a `component` to
    a specific indicator. An indicator can be used only if its corresponding
    `component` is specified. Supported indicators and their corresponding
    components are:
+
--
  `instance_has_master`::
      (Component `cluster_coordination`) Reports health issues regarding
      connections to the master node.

  `shards_availability`::
      (Component `data`) Reports health issues regarding shard assignments.

  `ilm`::
      (Component `data`) Reports health issues related to
      Indexing Lifecycle Management.

  `repository_integrity`::
      (Component `snapshot`) Tracks repository integrity and reports health issues
      that arise if repositories become corrupted.

  `slm`::
      (Component `snapshot`) Reports health issues related to
      Snapshot Lifecycle Management.
--

[[health-api-query-params]]
==== {api-query-parms-title}

`explain`::

>     (Optional, Boolean) If `true`, the response includes additional details that help explain the status of each non-green indicator.

Author: Added in [1c3d410](https://github.com/elastic/elasticsearch/pull/87139/commits/1c3d410179a8d779d0e81aebc9d7f65384acba1c)

[[health-api-desc]]
==== {api-description-title}

The health API returns the health status of an Elasticsearch cluster. It
returns a list of components that compose Elasticsearch functionality. Each
component's health is determined by health indicators associated with the
component.

Each indicator has a health status of: `green`, `unknown`, `yellow` or `red`. The indicator will
provide an explanation and metadata describing the reason for its current health status.

A component's status is controlled by the worst indicator status. The cluster's status
is controlled by the worst component status.

In the event that an indicator's status is non-green, a list of impacts may be present in the

> indicator result which detail the parts of a cluster that are negatively affected by the health issue.

Reviewer: ```suggestion
indicator result which detail the functionalities that are negatively affected by the health issue.
```

> Each impact carries with it a severity level, an area of the cluster that is affected, and a simple

Reviewer: ```suggestion
Each impact carries with it a severity level, an area of the system that is affected, and a simple
```

description of the impact on the system.

Some health indicators can determine the root cause of a health problem and prescribe a set of

> user actions that can be performed in order to improve the health status of an indicator. User

Reviewer: ```suggestion
user actions that can be performed in order to improve the health of the system. User
```
